test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1037,15 +1034,37 @@ UCS_TEST_P(test_ucp_am_nbx_closed_ep, rx_rts_am_on_closed_ep, "RNDV_THRESH=32K")
 UCP_INSTANTIATE_TEST_CASE(test_ucp_am_nbx_closed_ep)
 
 
-class test_ucp_am_nbx_eager_memtype : public test_ucp_am_nbx {
+class test_ucp_am_nbx_prereg : public test_ucp_am_nbx {
+public:
+    static void get_test_variants(variant_vec_t &variants)
+    {
+        add_prereg_variants(variants, test_ucp_am_nbx::get_test_variants);
```

Comment:
maybe move add_prereg_variants to "test_ucp_am_nbx_prereg" class 
and if need, inherit test_ucp_am_nbx_reply also from test_ucp_am_nbx_prereg
it's weird that we have some "prereg" logic leftover in test_ucp_am, that does not really use it